executor/README.md

```diff
@@ -0,0 +1,270 @@
+Cook Executor
+=============
+
+The cook executor is a custom executor written in python. It replaces the default command executor in order to enable a number of features for both operators and end users.
+
+For Users
+-----
+
+The cook executor supports a number of (completely optional) features beyond the standard Mesos command executor. It allows users to specify additional commands that run either before or after a job's primary command. This enables the use of existing utilities, and promotes sharing and reuse. It also provides an HTTP API that allows commands to expose progress updates and custom failure messages, as well as update environment variables.
+
+### Scheduler HTTP API
+
+#### `POST /rawscheduler`
+
+The job creation endpoint can be used to specify before and after commands, with the following keys:
+
+| key | type | description |
+|-----|------|-------------|
+| `before_commands` | array | An array of `command`s to run before all user specified commands. |
+| `after_commands` | array | An array of `command`s to run after all user specified commands. |
+
+Commands have the following structure:
+
+| key | type | description |
+|-----|------|-------------|
+| `value` | string | A string representing the command to be run, ex: `echo hello`. |
+| `guard` | boolean | An optional flag. If set, and the command returns a non-zero exit code, no further commands for the job will be executed. |
+| `async` | boolean | An optional flag. If set, the command runs asynchronously, in parallel with other future commands. It will be killed when all synchronous commands for a job are finished. |
+
+#### `GET /rawscheduler?job=:job`
+
+The job status endpoint can be used to view a job's progress, custom failure message, as well as the exit codes for all commands, by examining the following additional keys:
+
+| key | type    | description                              |
+| -------- | ------- | ---------------------------------------- |
+| `exit_code`     | integer | Status code when job specifies `command`  (will be absent if job has yet to finish). |
+| `before_exit_codes` | array of integers | Status codes for `before_commands`. |
```

Comment:
Please see comment above regarding using an `id` for command. Then we can return a map instead of an array of exit codes.